File: app/src/main/java/com/aar/app/wsp/data/room/UsedWordDataSource.java
package com.aar.app.wsp.data.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.aar.app.wsp.model.UsedWord;

import java.util.List;

@Dao
public interface UsedWordDataSource {

    @Query("SELECT * FROM used_words WHERE game_data_id=:gameDataId")
    List<UsedWord> getUsedWords(int gameDataId);

    @Query("SELECT COUNT(*) FROM used_words WHERE game_data_id=:gameDataId")
    int getUsedWordsCount(int gameDataId);

    @Query("UPDATE used_words SET answer_line=null, duration=0 WHERE game_data_id=:gameDataId")
    void resetUsedWords(int gameDataId);

    @Query("UPDATE used_words SET duration=:duration WHERE id=:usedWordId")
    void updateUsedWordDuration(int usedWordId, int duration);

    @Update
    void updateUsedWord(UsedWord usedWord);

    @Insert
    void insertAll(List<UsedWord> usedWords);

    @Query("DELETE FROM used_words WHERE game_data_id=:gameDataId")
    void removeUsedWords(int gameDataId);

    @Query("DELETE FROM used_words")
    void removeAll();

}

File: app/src/main/java/com/aar/app/wsp/data/room/UsedWordDataSource.kt
package com.aar.app.wsp.data.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.aar.app.wsp.model.UsedWord

@Dao
interface UsedWordDataSource {
    @Query("SELECT * FROM used_words WHERE game_data_id=:gameDataId")
    fun getUsedWords(gameDataId: Int): List<UsedWord>

    @Query("SELECT COUNT(*) FROM used_words WHERE game_data_id=:gameDataId")
    fun getUsedWordsCount(gameDataId: Int): Int

    @Query("UPDATE used_words SET answer_line=null, duration=0 WHERE game_data_id=:gameDataId")
    fun resetUsedWords(gameDataId: Int)

    @Query("UPDATE used_words SET duration=:duration WHERE id=:usedWordId")
    fun updateUsedWordDuration(usedWordId: Int, duration: Int)

    @Update
    fun updateUsedWord(usedWord: UsedWord)

    @Insert
    fun insertAll(usedWords: List<UsedWord>)

    @Query("DELETE FROM used_words WHERE game_data_id=:gameDataId")
    fun removeUsedWords(gameDataId: Int)

    @Query("DELETE FROM used_words")
    fun removeAll()
}